raft-tech/TANF-app
# Update cf os

## Summary of Changes
- Updated nginx build pack 
- Added script to automate app OS/FS upgrading for apps that dont have a regular deploy cycle
- Added workflows to circi for the upgrade
Pull request closes #2521 

## How to Test
The only way to test this is to try upgrading one of the existing frontend/backend apps in the `tanf-dev` space. You can use the `tdrs-deploy` command from `commands.sh` with appropriate arguments e.g. `tdrs-deploy sandbox`. Note sandbox has already been upgraded, so to see an actual upgrade you will want to run `tdrs-deploy` against a different environment. If the deploy succeeds, run `cf audit-stack` and verify that the environment you deployed specifies the `cflinuxfs4` OS/FS and that you can authenticate to the deployment.

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] Frontend apps successfully migrate
+ [ ] Backend apps successfully migrate
+ [ ] Other apps successfully migrate

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
